### PR TITLE
Ciphertrust integration

### PR DESCRIPTION
**Pull request by mayyagoldman**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: validation_config.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ select = [
     "DO100", "DO101", "DO102", "DO103", "DO104", "DO105", "DO106",
     "DS100",
     "SC100", "SC105", "SC106", "SC109",
-    "RM101", "RN103", "RM104", "RM105", "RM109", "RM113", "RM114",
```
Comment (YuvHayun) [inline validation_config.toml:36]: Don't remove it completely, move it to warning s,ection please.

**Review by YuvHayun — CHANGES_REQUESTED** [review]

**Review by omerKarkKatz — CHANGES_REQUESTED** [review]

File: Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py [hunk 2]
```diff
@@ -0,0 +1,1444 @@
...
+    Returns:
+        The human-readable date string.
+
+    """
+    if not iso_date:
+        return empty_value
+    try:
+        return datetime.strptime(iso_date, DATE_FORMAT).strftime(DATE_FORMAT_HR)
+    except ValueError:
+        return datetime.strptime(iso_date, DATE_FORMAT_NO_MS).strftime(DATE_FORMAT_HR)
```
[the remark this answers is not in the document]
Comment (mayyagoldman) [reply Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py:634]: Just a general exception to be printed to the console

File: Packs/ThalesCipherTrustManager/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  - **Create Local and External CAs**: Establish new Certificate Authorities that can be utilized for internal interfaces and services.
+  
+- **Digital Certificates Management**:
+  - **Issue and Install Certificates**: Manage the issuance and installation of server and client digital certificates along with certificate signing requests (CSR).
+  - **Revoke and Resume Certificates**: Provide functionality to revoke or resume certificates, specifically those signed by local CAs, ensuring compliance with security standards.
+
+For detailed description of each of the use cases, see the integration documentation.
+
+
+## Permissions
```
[the remark this answers is not in the document]
Comment (mayyagoldman) [reply Packs/ThalesCipherTrustManager/README.md:42]: the Permissions part?

[the remark this answers is not in the document]
Comment (omerKarkKatz) [reply Packs/ThalesCipherTrustManager/README.md:42]: yes.

File: Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py [hunk 4]
```diff
@@ -0,0 +1,1384 @@
...
+
+
+def ciphertrust_table_to_markdown_transform_data(data: dict, keys: list[str], keys_headers_mapping: dict,
```
Comment (omerKarkKatz) [inline Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py:573]: Please reomve this function now that we dont use it any more

File: Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py [hunk 5]
```diff
@@ -0,0 +1,1384 @@
...
+    active_cas = tableToMarkdown('Active CAs', [{
+        'Name': ca.get('name', ''),
+        'Subject': ca.get('subject', ''),
+        'Serial #': ca.get('serialNumber', ''),
+        'Activation': date_to_markdown(ca.get('notBefore', '')),
+        'Expiration': date_to_markdown(ca.get('notAfter', '')),
+        'Client Auth': 'Enabled' if ca.get('purpose', {}).get('client_authentication') else 'Disabled',
+        'User Auth': 'Enabled' if ca.get('purpose', {}).get('user_authentication') else 'Disabled'
+    } for ca in raw_response.get('resources', []) if ca.get('state') == 'active'], headerTransform=underscoreToCamelCase,
+                                 sort_headers=False)
```
Comment (omerKarkKatz) [inline Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py:657]: please add new lines after for loop, if/else or any other cluse statments

File: Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py [hunk 6]
```diff
@@ -0,0 +1,1384 @@
...
+    pending_cas = tableToMarkdown('Pending CAs', [{
```
Comment (omerKarkKatz) [inline Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py:658]:
```suggestion
    } for ca in raw_response.get('resources', []) if ca.get('state') == 'active'], headerTransform=underscoreToCamelCase,
                                 sort_headers=False)

    pending_cas = tableToMarkdown('Pending CAs', [{
```

File: Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py [hunk 7]
```diff
@@ -0,0 +1,1384 @@
...
+        'Name': ca.get('name', ''),
+        'Subject': ca.get('subject', ''),
+        'Created': date_to_markdown(ca.get('createdAt', '')),
+        'Fingerprint': ca.get('sha1Fingerprint', '')
+    } for ca in raw_response.get('resources', []) if ca.get('state') == 'pending'], headerTransform=underscoreToCamelCase,
+                                  sort_headers=False)
+    expired_cas = tableToMarkdown('Expired CAs', [{
```
Comment (omerKarkKatz) [inline Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py:665]:
```suggestion
        'Fingerprint': ca.get('sha1Fingerprint', '')
    } for ca in raw_response.get('resources', []) if ca.get('state') == 'pending'], headerTransform=underscoreToCamelCase,
                                  sort_headers=False)
                                  
    expired_cas = tableToMarkdown('Expired CAs', [{
```

File: Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py [hunk 8]
```diff
@@ -0,0 +1,1384 @@
...
+def group_list_markdown(data: dict) -> str:
+    skip_limit_total_hr = hr_skip_limit_to_markdown(data.get('skip', 0),
+                                                    data.get('limit', 0),
+                                                    data.get('total', 0),
+                                                    'Groups')
+
+    transformed_data = [{
+        'Name': group.get('name', ''),
+        'Defined By': 'System' if isinstance(group.get('app_metadata'), dict) and group.get('app_metadata').get(
+            'system') else 'User',
```
Comment (omerKarkKatz) [inline Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py:630]:
```suggestion
        'Defined By': 'System' if group.get('app_metadata', {}).get('system') ekse 'User',
```

**Review by omerKarkKatz — APPROVED** [review]

File: validation_config.toml [hunk 9]
```diff
@@ -43,8 +43,8 @@ select = [
     "IT100", "IT101", "IT102", "IT103",
     "MR100",
 ]
-warning = ["DS107", "LO107", "RN103"]
 
+warning = ["DS107", "LO107", "RN103" ]
```
Comment (YuvHayun) [inline validation_config.toml:47]: Was already merged in another pr, revert changes on this file.

**Review by YuvHayun — APPROVED** [review]

**Pull request merged**